### DISCUSSION TITLE
Running parallel batches for a course

### DISCUSSION THREAD

**Discussion by coolbung** [Sunbird-Ed/Community#155] [Sunbird ED configuration discussion]
Currently the Sunbird portal does not allow creation of course batches that run in parallel i.e. there cannot be 2 batches with overlapping dates. Is this a restriction only from the UI side, or does the API also enforce this ?

cc @baljeetsingh @vinod303 @komalm

**Comment by pramodkvarma** [comment]
My gut feel is that there is no backend logic checking batch dates. Most tracking in database etc is probably based on course ID and batch ID. Maybe there are some silly issues with multiple concurrent batches due to DB look up etc.

Adding @vrayulu to thread.

@coolbung why don't you test this in sandbox via APIs? Easier that way! Test creation, subscription, progress tracking, etc.

**Reply by coolbung** [reply]
Yes, @vinod303 @komalm on it (test and find out). Started the thread to document the outcome.

**Reply by vrayulu** [reply]
The API doesn't have any logic to not allow creating multiple active batches for a course. And all tracking is specific to a course and a batch.

One thing to be checked is what logic does enrolment API has for users enrolment into more than one batch of a course (@maheshkumargangula ?). 

@coolbung - what does your use case need: allow users to enrol into multiple batches of a course or not, and why?

**Reply by coolbung** [reply]
Right now we don't see a use case for the same student to enroll to multiple batches. However multiple parallel batches are needed to track multiple user cohorts who are doing the same course in parallel. 

@komalm @xitija -- please confirm that the batch creation API does not have any limitations.

**Comment by surendrasinghs** [comment]
I can imagine how having multiple batches running in parallel might have created confusion for the end-user. At the same time, I can also imagine the need for having different batches (with the same dates). Can you please help me understand what's the use case you are trying to enable?

I think that instead of restricting creation, the system should warn/suggest to the batch creator that "a similar batch already exists". Let the batch creator decide from there on... 
Similarly, when enrolling in a course, the system could show similar batches so that learners can enrol in the intended batch.

**Reply by baljeetsingh** [reply]
This NGO working with several different bodies to educate them on various aspect they have a set scheule set for them on which a learning journey begins and end.

They also have these course actually run on invite only course, same course used for multiple batches.

**Reply by kameshbhr** [reply]
Yes, there was a similar thought earlier in DIKSHA context also - same course having different batches for teachers of different districts.
So having multiple batches for the same course seems a valid need for multiple use cases. 
However, I don't think it would make sense for a user to be able to enroll into two batches of the same course running parallelly. That needs to be avoided.  Also, if the course is self-enrollment, and a user sees two batches open to enroll during the same time, it would be confusing. For invite only courses, this will not be a problem.

**Reply by baljeetsingh** [reply]
Yes, we are going to use and invite to specific users, these are not open courses.

**Comment by chitranshu-keshav** [comment]
@coolbung was the tests concluded? Can you please share the document link on this thread and mark it as answered

cc: @Krishnaj20
